lib/xchain/connect/connect.go

```diff
@@ -51,6 +53,21 @@ func (c Connector) Backend(chainID uint64) (*ethbackend.Backend, error) {
 	return ethbackend.NewBackend(chain.Name, chainID, chain.BlockPeriod, cl)
 }
 
+// GetPortal returns an OmniPortal binding for the portal at the provided address.
+func (c Connector) GetPortal(chainID uint64, backend bind.ContractBackend) (*bindings.OmniPortal, error) {
```

Comment:
xconnector isn't used in monitor, it is only used in scripts, lets keep it that way. please remove this.